### DISCUSSION TITLE
Passing UploadFile objects into a StreamingResponse closes it in v0.106.0 but not v0.105.0

### DISCUSSION THREAD

**Discussion by adrwz** [fastapi/fastapi#10856] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
async def upload_file_event(files: List[UploadFile]):
    print("2", files[0].filename, files[0].file._file.closed)

    # ... more code ...


@router.post("/upload_session_file")
async def upload_file(files: List[UploadFile] = File(...)):
    print("1", files[0].filename, files[0].file._file.closed)

    try:
        return StreamingResponse(
            upload_file_event(files),
            media_type="text/event-stream",
        )
    except Exception as error:
        logging.error(error)
        raise HTTPException(status_code=500, detail="Internal Server Error")
```

### Description

Hitting this endpoint with an uploaded file on v0.105.0 will print:

```
1 filename false
2 filename false
```

Hitting this endpoint with an uploaded file on v0.106.0 will print:

```
1 filename false
2 filename true
```

Unfortunately this means I'm unable to upload files in a StreamingResponse with fastapi>=0.106.0

### Operating System

macOS

### FastAPI Version

0.105.0

### Pydantic Version

2.5.3

### Python Version

3.11.4

**Comment by Kludex** [comment, marked as answer]
I've created https://github.com/tiangolo/fastapi/issues/10857 from this discussion.
